### ISSUE TITLE
[Request for External] page 9816 "Permission Set by User"

### ISSUE THREAD

**Issue by panas78** [microsoft/ALAppExtensions#23962]
Could you please change the Access scope 

to the following variable to protected

```
   protected var
        AllUsersHavePermission: Boolean;
```

and also to the following procedure
```
protected procedure SetUserPermission(UserSecurityID: Guid; UserHasPermission: Boolean)
```

**Comment by stkillen** [comment]
> Could you please change the Access scope
> 
> to the following variable to protected
> 
> ```
>    protected var
>         AllUsersHavePermission: Boolean;
> ```
> 
> and also to the following procedure
> 
> ```
> protected procedure SetUserPermission(UserSecurityID: Guid; UserHasPermission: Boolean)
> ```



What scenario are you trying to support?

The `SetUserPermission` procedure could be implemented in your own page extension - and what is the reason for changing the `AllUsersHavePermission `access scope?

**Comment by panas78** [comment]
Here is our page extension.
Our goal was to leave my UserId unaffected in case for instance someone selects to remover SUPER from All users.

```
pageextension 52004611 "Permission Set by User" extends "Permission Set by User"
{
    layout
    {

        modify(AllUsersHavePermission)
        {
            Visible = false;
            Enabled = false;
        }

        addafter(AllUsersHavePermission)
        {
            field(AllUsersHavePermission2; AllUsersHavePermission)
            {
                ApplicationArea = Basic, Suite;
                Caption = 'All Users';
                ToolTip = 'Specifies that the permission set will be assigned to all users.';

                trigger OnValidate()
                begin
                    SetAllUsersColumnPermission(AllUsersHavePermission);
                    CurrPage.Update(false);
                end;
            }
        }

    }

    local procedure SetAllUsersColumnPermission(UserHasPermission: Boolean)
    var
        User: Record User;
        EnvironmentInfo: Codeunit "Environment Information";
    begin

        if EnvironmentInfo.IsSaaS then
            User.SetFilter("License Type", '<>%1', User."License Type"::"External User");
        if not UserHasPermission then
            User.SetFilter("User Security ID", '<>%1', UserSecurityId());
        if User.FindSet() then
            repeat
                SetUserPermission(User."User Security ID", UserHasPermission);
            until User.Next() = 0;

    end;

}

```

**Comment by JesperSchulz** [comment]
Fix in 23.0.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 11810.
